_posts/2016-05-13-rustup.md

```diff
@@ -0,0 +1,553 @@
+        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fca26ae6000)
+        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fca268cf000)
+        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fca26506000)
+        /lib64/ld-linux-x86-64.so.2 (0x000056104c935000)
+        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fca261fd000)
+```
+
+To compile for musl instead call `cargo` with the argument
+`--target=x86_64-unknown-linux-musl`. If we just go ahead and try that
+we'll get an error:
+
+```
+rust:~/hello$ cargo run --target=x86_64-unknown-linux-musl
+   Compiling hello v0.1.0 (file:///home/rust/hello)
+error: can't find crate for `std` [E0463]
+error: aborting due to previous error
+Could not compile `hello`.
+...
+
+```
+
+The error tells us that the compiler can't find `std`. That is of
+course because we haven't installed it.
+
+To start cross-compiling, you need to acquire a standard library for the target
+platform. Previously, this was an error-prone, manual process &mdash; queue those
+blog posts I mentioned earlier. But with rustup, it's just part of the usual
+workflow:
+
+```
+rust:~/hello$ rustup target add x86_64-unknown-linux-musl
+info: downloading component 'rust-std' for 'x86_64-unknown-linux-musl'
+info: installing component 'rust-std' for 'x86_64-unknown-linux-musl'
+rust:~/hello$ rustup show
+installed targets for active toolchain
+--------------------------------------
+
+x86_64-unknown-linux-gnu
+x86_64-unknown-linux-musl
+
+active toolchain
+----------------
+
+stable-x86_64-unknown-linux-gnu (default)
+rustc 1.8.0 (db2939409 2016-04-11)
+```
+
+So I'm running the 1.8 toolchain for Linux on 64-bit x86, as indicated by the
+`x86_64-unknown-linux-gnu` target triple, and now I can also target
+`x86_64-unknown-linux-musl`. Neat. Surely we are ready to build a slick
+statically-linked binary we can release into the cloud. Let's try:
+
+```
+rust:~/hello$ cargo run --target=x86_64-unknown-linux-musl
+   Compiling hello v0.1.0 (file:///hello)
+     Running `target/x86_64-unknown-linux-musl/debug/hello`
+Hello, world!
+```
+
+And that... just worked! Run `ldd` on it for proof that it's the real
+deal:
+
+```
+rust:~/hello$ ldd target/x86_64-unknown-linux-musl/debug/hello
+        not a dynamic executable
+```
+
+Now take that `hello` binary and copy it to any x86_64 machine running Linux and
+it'll run just fine.
+
+For more advanced use of musl consider [rust-musl-builder], a Docker
+image set up for musl development, which helpfully includes common C
+libraries compiled for musl.
+
+[rust-musl-builder]: https://github.com/emk/rust-musl-builder
+
+## Example: Running Rust on Android
+
+One more example. This time building for Android, from Linux, i.e.,
+`arm-linux-androideabi` from `x86_64-unknown-linux-gnu`.  This can also be done
+from OS X or Windows, though on Windows the process is [less smooth right now][lsrt].
```

Comment:
I think we should drop the less smooth link -- the text there isn't fully polished.